packages/test/test-version-utils/README.md

```diff
@@ -1,26 +1,137 @@
 # @fluidframework/test-version-utils
 
-Provide Fluid APIs in different version for testing.
-Separate into Loader/ContainerRuntime/DataObject+DDS category for now.
-- getLoaderApi
-- getContainerRuntimeApi
-- getDataRuntimeApi
+This is a package for writing and setting up Fluid end to end tests using `mocha` that will generate variants with
+a specific driver and different version combinations of Fluid API between layers via `TestObjectProvider` provided
+to the test. The different layers are loader, driver, container runtime and data runtime (includes DDS).  Version
+combinations and driver selection can be controlled via the command line for test set up using the provided
+ `describe*` functions.  For advanced usage, test suite can bypass the setup and command line and directly call our
+exports to get the versioned Fluid APIs.
+
+## Versioned combination test generation
+
+### Layer version combinations
+
+Similar to `mocha`'s `describe`, this package provide various `describe*` functions that will generate variants with
+a specific driver and different version combinations of Fluid API between layers. All possible layer combinations that
+are generated (empty entries are current versions):
+
+| Compat Kind         | Loader | Driver | Container Runtime | Data Runtime |
+| ------------------- | ------ | ------ | ----------------- | ------------ |
+| None                |        |        |                   |              |
+| Loader              |  old   |        |                   |              |
+| Driver              |        |  old   |                   |              |
+| ContainerRuntime    |        |        | old               |              |
+| DataRuntime         |        |        |                   | old          |
+| NewLoader           |        |  old   | old               | old          |
+| NewDriver           |  old   |        | old               | old          |
+| NewContainerRuntime |  old   |  old   |                   | old          |
+| NewDataRuntime      |  old   |  old   | old               |              |
+
+### Mocha test setup with layer version combinations
+
+There are three compat `describe*` to generate different combinations, depending of the need of the tests
+
+`describeFullCompat`: generate test variants with compat combinations that varies the version for all layers.
+
+- Used for tests that exercise all layers and will benefits compat combinations of all layers.
+
+`describeLoaderCompat`: generate test variants with compat combinations that only varies the loader version.
+
+- Use for tests that targets the loader layer, and don't care about compat combinations of other layers.
+- Test combination generated: [CompatKind.None, CompatKind.Loader]
```

Comment:
Why not also `NewLoader`?

Reply:
I see that currently it has limited use in examples that doesn't use the `TestObjectProvider` to take advantage of the legacy runtime anyway.   So it doesn't have any benefits with `NewLoader`, as the runtime will always be "new"

Reply:
Gotcha.  I opened #9581, seems like we should be testing new loader if we're ever targeting loader compatibility specifically.